File: PresentationTest/PopAdTransitioningDelegate.m
//
//  PopAdTransitioningDelegate.m
//  PresentationTest
//
//  Created by Minxin Guo on 5/9/16.
//  Copyright © 2016 aisle411. All rights reserved.
//

#import "PopAdTransitioningDelegate.h"
#import "PopAdPresentationController.h"
#import "PopAdAnimatedTransitioning.h"

@implementation PopAdTransitioningDelegate
-(UIPresentationController *)presentationControllerForPresentedViewController:(UIViewController *)presented presentingViewController:(UIViewController *)presenting sourceViewController:(UIViewController *)source {
    PopAdPresentationController *presentationController = [[PopAdPresentationController alloc] initWithPresentedViewController:presented presentingViewController:presenting];
    return presentationController;
}

-(id<UIViewControllerAnimatedTransitioning>)animationControllerForPresentedController:(UIViewController *)presented presentingController:(UIViewController *)presenting sourceController:(UIViewController *)source {
    PopAdAnimatedTransitioning *animationController = [[PopAdAnimatedTransitioning alloc] init];
    animationController.isPresentation = true;
    return  animationController;
}

-(id<UIViewControllerAnimatedTransitioning>)animationControllerForDismissedController:(UIViewController *)dismissed {
    PopAdAnimatedTransitioning *animationController = [[PopAdAnimatedTransitioning alloc] init];
    animationController.isPresentation = false;
    return  animationController;
}

@end

File: PresentationTest/PopAdTransitioningDelegate.swift
//
//  PopAdTransitioningDelegate.swift
//  PresentationTest
//
//  Created by Minxin Guo on 5/5/16.
//  Copyright © 2016 aisle411. All rights reserved.
//

import UIKit

class PopAdTransitioningDelegate: NSObject, UIViewControllerTransitioningDelegate {
    func presentationControllerForPresentedViewController(presented: UIViewController, presentingViewController presenting: UIViewController, sourceViewController source: UIViewController) -> UIPresentationController? {
        let presentationController = PopAdPresentationController(presentedViewController: presented, presentingViewController: presenting)
        return presentationController
    }
    
    func animationControllerForPresentedController(presented: UIViewController, presentingController presenting: UIViewController, sourceController source: UIViewController) -> UIViewControllerAnimatedTransitioning? {
        let animationController = PopAdAnimatedTransitioning()
        animationController.isPresentation = true
        return animationController
    }
    
    
    func animationControllerForDismissedController(dismissed: UIViewController) -> UIViewControllerAnimatedTransitioning? {
        let animationController = PopAdAnimatedTransitioning()
        animationController.isPresentation = false
        return animationController
    }
}
